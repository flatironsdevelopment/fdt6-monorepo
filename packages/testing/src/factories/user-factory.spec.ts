import faker from './faker'
import { UserFactory } from './user-factory'

describe('UserFactory', () => {
  beforeEach(async () => {
    faker.create.resetSeed!()
  })

  it('should return a mock user', () => {
    expect(UserFactory.build()).toMatchSnapshot()
  })

  it('should return the same user', () => {
    const firstUser = UserFactory.build()
    faker.create.resetSeed!()
    const secondUser = UserFactory.build()

    expect(`${firstUser}`).toBe(`${secondUser}`)
  })
  it('should return a different user', () => {
    const firstUser = UserFactory.build()
    const secondUser = UserFactory.build()

    expect(firstUser).not.toBe(secondUser)
  })
})
