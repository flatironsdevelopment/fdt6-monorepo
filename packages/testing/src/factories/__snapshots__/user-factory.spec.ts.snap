// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserFactory should return a mock user 1`] = `
{
  "email": "Alek45@gmail.com",
  "emailVerified": true,
  "firstName": "Connor",
  "id": "pZIV07iZ9e5obolFoWxQpjGw",
  "lastName": "Thiel",
  "phoneNumber": "519-378-7983 x0658",
  "phoneNumberVerified": true,
}
`;
